src/main/java/com/hcl/appscan/jenkins/plugin/scanners/StaticAnalyzer.java

```diff
@@ -32,29 +32,48 @@ public class StaticAnalyzer extends Scanner {
         
         private boolean m_openSourceOnly;
         private boolean m_sourceCodeOnly;
+        private String m_scanSpeed;
         
         @Deprecated
         public StaticAnalyzer(String target){
             this(target,false);
         }
         
-        public StaticAnalyzer(String target, boolean hasOptions, boolean openSourceOnly, boolean sourceCodeOnly){
+        public StaticAnalyzer(String target, boolean hasOptions, boolean openSourceOnly, boolean sourceCodeOnly, String scanSpeed){
             super(target, hasOptions);
             m_openSourceOnly=openSourceOnly;
             m_sourceCodeOnly=sourceCodeOnly;
+            m_scanSpeed=scanSpeed;
         }
         
 	@DataBoundConstructor
 	public StaticAnalyzer(String target,boolean hasOptions) {
 		super(target, hasOptions);
                 m_openSourceOnly=false;
                 m_sourceCodeOnly=false;
+                m_scanSpeed="";
 	}
 
 	@Override
 	public String getType() {
 		return STATIC_ANALYZER;
 	}
+
+    	@DataBoundSetter
+   	public void setScanSpeed(String scanSpeed) {
+            	m_scanSpeed = scanSpeed;
+    	}
+
+    	public String getScanSpeed() {
+        	return m_scanSpeed;
+    	}
+
+    	public String checkScanSpeed(String scanSpeed) {
+        	if (m_scanSpeed != null) {
+            	return m_scanSpeed.equalsIgnoreCase(scanSpeed) ? "true" : "false";
+        		}
+        	return null;
+    	}
```

Comment:
We need this method should return null if the m_scanspeed is null(which will be while doing the upgrade) because, in the jelly file, we are handling the default behavior with this method: "checked=**"${(instance.checkScanSpeed('fast') == null) ? 'true' : (instance.checkScanSpeed('fast'))}"**  ".

Reply:
If I don't return null, then while doing the upgrade testing, if I open the pre-configured job...I found that none of the scan speeds will be chosen.